plugins/spectate.cpp

```diff
@@ -31,25 +33,102 @@ using namespace df::enums;
 
 DFHACK_PLUGIN("spectate");
 DFHACK_PLUGIN_IS_ENABLED(enabled);
+bool dismiss_pause_events = false;
 bool following_dwarf = false;
+df::unit* our_dorf = nullptr;
 void* job_watched = nullptr;
 int32_t timestamp = -1;
 REQUIRE_GLOBAL(world);
 REQUIRE_GLOBAL(ui);
+REQUIRE_GLOBAL(pause_state);
+REQUIRE_GLOBAL(d_init);
+
+// todo: implement as user configurable variables
+#define tick_span 50
+#define base 0.99
+Pausing::AnnouncementLock* pause_lock = nullptr;
+bool lock_collision = false;
 
 command_result spectate (color_ostream &out, std::vector <std::string> & parameters);
 
 DFhackCExport command_result plugin_init (color_ostream &out, std::vector <PluginCommand> &commands) {
     commands.push_back(PluginCommand("spectate",
                                      "Automated spectator mode.",
-                                     spectate));
+                                     spectate,
+                                     false,
+                                     ""
+                                     " spectate\n"
+                                     "    displays plugin status\n"
+                                     " spectate enable\n"
+                                     " spectate disable\n"
+                                     " spectate auto-unpause\n"
+                                     "    toggle auto-dismissal of game pause events. e.g. a siege event pause\n"
+                                     "\n"));
```

Comment:
docs go in the .rst files. in-tree plugins should not have any docs here. too easy to let them get out of sync, and this string won't be used anywhere anyway. the rendered help from the .rst file takes precedence.

Reply:
Oh ok. That's swell

Reply:
current documentation standards are [here](https://dfhack--2247.org.readthedocs.build/en/2247/docs/Documentation.html#documentation-standards)